src/Tasks/Microsoft.NET.Build.Tasks/GenerateShims.cs

```diff
@@ -113,20 +113,20 @@ protected override void ExecuteCore()
 
                 try
                 {
+                    var windowsGraphicalUserInterface = runtimeIdentifier.StartsWith("win") && "WinExe".Equals(OutputType, StringComparison.OrdinalIgnoreCase);
```

Comment:
@nguerrera @peterhuene 
sorry for the churn. The feedback loop of tests is slow.

I realize I do need to recalculate `WindowsGraphicalUserInterface`, since `WindowsGraphicalUserInterface` depends on `UseApphost` which will always be false in dotnet tools.